R/descriptives.R

```diff
@@ -1181,6 +1202,64 @@ Descriptives <- function(jaspResults, dataset, options) {
   return(p)
 }
 
+.descriptivesDotPlots <- function(dataset, options, variable){
+  
+  
+  if (options$splitby != "" ) {
+    # return a collection
+    levelsSplitFactor <- names(dataset)
+    
+    plotContainer <- createJaspContainer(title = variable)
+    plotContainer$dependOn(optionContainsValue = list(variables = variable))
+    
+    for (level in levelsSplitFactor)
+      plotContainer[[level]] <- .descriptivesDotPlots_SubFunc(dataset = dataset[[level]], variable = variable, title = level)
+    
+    return(plotContainer)
+    
+  }else{
+    
+    dotplot <- .descriptivesDotPlots_SubFunc(dataset = dataset, variable = variable, title = variable)
+    dotplot$dependOn(optionContainsValue = list(variables = variable))
+    
+    return(dotplot)
+  }
+}
+
+.descriptivesDotPlots_SubFunc <- function(dataset, variable, title){
+  
+  dotplot <- createJaspPlot(title = title)
+  
+  x <- na.omit(dataset[[.v(variable)]])
+  
+  if (is.factor(x)){
+    tb <- as.data.frame(table(x))
+    p <- ggplot2::ggplot(data = data.frame(x = x), ggplot2::aes(x = x)) + 
```

Comment:
This plotting code is nearly identical for both cases, except for this part. I'd change this into something like:
```r

  if (is.factor(x)) {
    tb <- table(x)
    df <- as.data.frame(tb)
    scale <- ggplot2::scale_x_discrete(limits = factor(tb[, 1]))
  } else {
    df <- data.frame(x = x)
    scale <- NULL
  }

  p <- ggplot2::ggplot(data = df, ggplot2::aes(x = x)) + 
    ggplot2::geom_dotplot(binaxis = 'x', stackdir='up') + 
    ggplot2::xlab(variable) +
    ggplot2::ylab(NULL) +
    scale +
    jaspGraphs::geom_rangeframe(sides = "b") +
    jaspGraphs::themeJaspRaw() +
    ggplot2::theme(
      axis.ticks.y = ggplot2::element_blank(),
      axis.title.y = ggplot2::element_blank(),
      axis.text.y  = ggplot2::element_blank()
    )
```
also, `jaspGraphs::geom_rangeframe(sides = "b") + jaspGraphs::themeJaspRaw()` is preferred over `jaspGraphs::themeJasp` (this changed).